client/state/purchases/test/reducer.js

```diff
@@ -33,27 +33,29 @@ describe( 'reducer', () => {
 	} );
 
 	test( 'should return an object with the list of purchases when fetching completed', () => {
+		const siteId2 = '2828828282';
 		let state = reducer( undefined, {
 			type: PURCHASES_USER_FETCH_COMPLETED,
 			purchases: [
-				{ ID: '1', blog_id: siteId, user_id: userId },
+				{ ID: '1', blog_id: siteId2, user_id: userId },
 				{ ID: '2', blog_id: siteId, user_id: userId },
 			],
 		} );
 
 		state = reducer( state, {
 			type: PURCHASES_SITE_FETCH_COMPLETED,
 			purchases: [
-				{ ID: '2', blog_id: siteId, user_id: userId },
+				{ ID: '2', blog_id: siteId, user_id: userId, other: true },
```

Comment:
Add a property to (the already-existing) id 2 so that we can prove it got updated by the site fetch.